### PR TITLE
Removed dead codes in the SwitchAnalyzer.cpp File

### PR DESCRIPTION
**Pull request by mrSeoD**
A fix for the #5182 issue @0xdaryl 

Two things fixed: 
1. if(branchTable) statement in the addTableBlock() method is deleted as branchTable = 0 all the time, meaning this statement will never execute. 
2. "branchTableSymRef" pointer is deleted as it's not used as a result of #1 action. 

If there are more things to fix, please let me know!

### CODE REVIEW

File: compiler/optimizer/SwitchAnalyzer.cpp [hunk 1]
```diff
@@ -1257,8 +1255,6 @@ TR::Block *TR::SwitchAnalyzer::addTableBlock(SwitchInfo *dense)
 
    CASECONST_TYPE upperBound = dense->_max - dense->_min;
 
-   TR::SymbolReference *branchTableSymRef = NULL;
-
    int32_t branchTable = 0;
```
Comment (andrewcraik) [inline compiler/optimizer/SwitchAnalyzer.cpp:1258]: Is this variable also always zero and unused once the dead code you have written is cleaned up? If so could you tidy this up too?

**Comment by andrewcraik** [comment]
@mrSeoD can you please make sure you have signed the ECA and have signed your commits per the contribution guidelines? Please also update the copyright header on the files to make sure the second year is 2020.

**Comment by andrewcraik** [comment]
When you sign your commits please squash the changes as the change is small enough to not need 2 commits. Thank you for making your first contribution to the project!

**Comment by andrewcraik** [comment]
@mrSeoD  just wanted to check how you were going with signing the ECA and addressing the review comment above? Thank you again for taking the time to contribute to Eclipse OMR!

**Comment by mrSeoD** [comment]
Hi @andrewcraik, thank you so much for the update! I apologize for such a late response as I was busy all week. I will make sure to sign ECA and make changes that you asked me by tomorrow if that's okay with you. As a beginner, I am not sure how to "squash" the commits. Is there any advice you could provide? Thanks a lot again.

**Comment by andrewcraik** [comment]
The project has documentation on standard git operations and links to resources in docs/GitCrashCourse.md so I'd start there and some web searches should help you understand what you need to do.

**Comment by mrSeoD** [comment]
Hi, I worked on getting rid of the variable " branchTable" and changing the date from 2019 to 2020. I tried signing the ECA several times, but it's giving me an error: "Error, An error has occurred and your Eclipse Contributor Agreement was not submitted. (ECA-04)" Also, I tried squashing following tutorials online, but I get an error "E163: There is only one file to edit." I searched both of them online, but I wasn't able to find any solutions. I would appreciate your help!

**Comment by 0xdaryl** [comment]
> I tried signing the ECA several times, but it's giving me an error...

Googling the error reveals that others that encountered this problem in the past found it might be an Eclipse server issue.  I suggest trying again in the next day or so.  If it still persists for you, you may need to contact the Eclipse Foundation for support.

>  I tried squashing following tutorials online, but I get an error "E163: There is only one file to edit."

I believe this is a vim error not related to git.  Have you read through our git crash course [1]?  If so, can you be more specific about the steps you are taking to squash your commits (i.e., the commands you issued and the output)?  Otherwise it is difficult to see where your problem is.

Another reference you might find helpful to squash commits is [2].

[1] https://github.com/eclipse/omr/blob/master/doc/GitCrashCourse.md
[2] https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git

**Comment by mrSeoD** [comment]
@0xdaryl Thank you so much for your help! I tried signing the ECA again today, but it's giving me an error still. I will try later in the day or tomorrow. 

For the squashing, these are the commands I ran:

git rebase -i HEAD~3
(changed first two commits to "squash" and left the last one as "pick")
(pressed esc keyboard button)
(typed :wp) -- to save and close the editor
Then the error of ""E163: There is only one file to edit." comes up. 

Thanks your help!!

**Comment by Leonardo2718** [comment]
@mrSeoD 

> I tried signing the ECA again today, but it's giving me an error still. I will try later in the day or tomorrow.

It looks like you've correctly signed the ECA, but you're missing the "Signed-off-by" footer (see https://github.com/eclipse/omr/blob/master/CONTRIBUTING.md#commit-guidelines).

> For the squashing, these are the commands I ran:
>
> git rebase -i HEAD~3
> (changed first two commits to "squash" and left the last one as "pick")
> (pressed esc keyboard button)
> (typed :wp) -- to save and close the editor
> Then the error of ""E163: There is only one file to edit." comes up.

Just to make sure you got the order right, from top to bottom, you have to pick the _first_ commit and squash the _last two_ commits, something like:

```
# git rebase -i
pick commit-sha 1
squash commit-sha 2
squash commit-sha 3
```

**Comment by mrSeoD** [comment]
@Leonardo2718 Sorry for a late reply; I was busy the whole week. I really appreciate your help!! I will make sure to have the "Signed-off-by" footer. I actually did the opposite way and put the "pick" on the last one; now everything makes sense!

**Comment by mrSeoD** [comment]
@Leonardo2718 I've added the signed-off-by footer by stating:
git commit -m "For the footer Signed-off-by: David Seo seodavid221@gmail.com"
I would appreciate if you could confirm that this is the right way. 

For squashing, I tried this way: 
pick 0924bb003 ... 
squash 4d6a8a71 ... 
squash ecb50f663 ...
squash 9babca255 ...

(Preseed "esc" on the keyboard --> tried typing both ":wp" and ":wp!"; however, both are resulting in the same error "E163: There is only one file to edit." 
I see online about this problem is related to vim, but I am not sure how this is related to vim when I'm not using it. 

Thanks a lot for your help!

**Comment by andrewcraik** [comment]
@mrSeoD unfortunately that is not quite right. Every commit in your PR needs to be signed. The easiest way to do this is to add the -s option to the git commit command. The signed off text needs to appear on its own line separated from the main commit message by a blank line.

Further, for this change can you please squash the commits into a single commit - the change is small enough that we only need a single change in the project history rather than all the intermediate states from when you were developing the change.

Thanks!

**Comment by 0xdaryl** [comment]
Unless you've explicitly set the editor for git to use (via `git config --global core.editor <my favourite editor>`) git will use whatever the shell editor is defined in `VISUAL` or `EDITOR`.  If all else fails, it will use `vi`.

Your error reported above is a `vi `error, and likely caused because you should be using `wq` to write-and-quit rather than `wp` (write and switch to previous file, which doesn't exist, and hence the error).

All of this advice is already covered in the Git Crash Course doc [1] that I encourage you to read.

[1] https://github.com/eclipse/omr/blob/master/doc/GitCrashCourse.md

**Comment by mrSeoD** [comment]
@andrewcraik Thank you so much for your help! I just wanted to make sure that I can write like "git commit -s" and it will sign the "signed-off-by:" message to all the commits by itself? If so, do I need to make a new commit to do it and squash everything? Or, is it fine to just run the "git commit -s" command right away. 

@0xdaryl Thank you so much for the explanation! I will make sure to use "wq" instead of "wp"! I will make sure to run it once I get a confirmation from @andrewcraik.

**Comment by andrewcraik** [comment]
@mrSeoD I would suggest collapsing the commits and then doing a git commit --amend -s on the single commit to add the sign off line to the commit message in the right format. A web search for commit signing will give you a lot of information on how signing works.

**Comment by andrewcraik** [comment]
@mrSeoD was wondering if you were having any issues we could help with? This is a good clean-up that we would be keen to accept once the review comments are addressed. If you need more interactive help drop by the project Slack workspace - there are plenty of us hanging out on there who would be happy to help answer questions.

**Comment by mrSeoD** [comment]
@andrewcraik I apologize not being able to update you sooner :(( I was busy preparing and settling in with my college. I will work on it today and give you updates!

**Comment by mrSeoD** [comment]
@andrewcraik I just tried squashing everything and ran the "git commit --amend -s" to sign the footer!! So so thankful for all of you guys' help. I would appreciate if you could confirm that things look good!

The following is what I got from running on the command prompt:

C:\Users\seoda\omr\compiler\optimizer>git rebase -i HEAD~4                                                              
[detached HEAD 89480f0e6] Removed dead codes from SwitchAnalyzer.cpp file                                                
Date: Wed May 13 15:09:53 2020 -0700                                                                                    
1 file changed, 3 insertions(+), 15 deletions(-)                                                                       
Successfully rebased and updated detached HEAD.
                                                                                                                                                                                                C:\Users\seoda\omr\compiler\optimizer>git commit --amend -s                                                             
[detached HEAD 180035abc] Removed dead codes from SwitchAnalyzer.cpp file
Date: Wed May 13 15:09:53 2020 -0700
1 file changed, 3 insertions(+), 15 deletions(-)

**Comment by andrewcraik** [comment]
@mrSeoD did you force push? I still see 4 commits in the PR...

**Comment by mrSeoD** [comment]
@andrewcraik I'm so sorry I didn't push it. I just pushed it and the commit is one now!

**Comment by andrewcraik** [comment]
@mrSeoD something looks like it went wrong - you have conflicting changes in gc/base/j9mm.tdf and i don't think you intend to have those in your commit. Can you have a look and address that problem to remove the conflict?

**Comment by mrSeoD** [comment]
Hi @andrewcraik, I tried reading and understanding the text that got added, but was not sure of the meanings: 

<<<<<<< dead_code_edit_1
=======

TraceEntry=Trc_MM_MemorySubSpaceUniSpace_getHeapFreeMaximumHeuristicMultiplier Overhead=1 Level=1 Group=resize Template="Trc_MM_MemorySubSpaceUniSpace_getHeapFreeMaximumHeuristicMultiplier Maximum free multiplier = %zu"
TraceEntry=Trc_MM_MemorySubSpaceUniSpace_getHeapFreeMinimumHeuristicMultiplier Overhead=1 Level=1 Group=resize Template="Trc_MM_MemorySubSpaceUniSpace_getHeapFreeMinimumHeuristicMultiplier Minimum free multiplier = %zu"
>>>>>>> master

I am guessing this text got added as a result of the commit for some reason. should I delete them and push the changes again?

**Comment by andrewcraik** [comment]
@mrSeoD looks like your rebase went a bit wrong. I think you should remove those changes from your commit and then we can continue the review.

**Comment by mrSeoD** [comment]
@andrewcraik I see that certain values in the file have changed as you can see in the picture: 

Screen Shot 2020-08-31 at 1 34 08 PM

Also, I noticed that the Scanvenger.cpp file got changed as well 
Screen Shot 2020-08-31 at 1 34 54 PM

I did not touch those files at all, so I am a little confused how they were changed. 
Could those file be changed during rebasing?

To fix the problem, should I just manually change those files and re-commit/push?

**Comment by andrewcraik** [comment]
You need to make sure your change is based on the latest head and undo any changes to files that were not intended and force push to fix up the PR so we can continue.

**Comment by mrSeoD** [comment]
@andrewcraik Got it! Thank you for the clarification. Before moving onto undoing any changes, I realized that "Aleksandar Micic <amicic@ca.ibm.com>" made changes to the Scavenger.cpp file (maybe to the j9mm.tdf file as well but not sure):

Screen Shot 2020-09-02 at 12 46 22 PM

Should I still continue and undo these changes? 
Lastly, when you say "undo," do you mean to manually get rid of changes or is there anyway we can do through git commands?
Thanks a lot again!

**Comment by andrewcraik** [comment]
@mrSeoD the only changes in your commit should be the changes you are wanting to make - anything else is a change to the source and not one you intended to make. The files you are not intending to change should match the files at the revision you are based on.

**Comment by mrSeoD** [comment]
@andrewcraik I apologize for a late replay; I've been busy with school works. Then, should I search up the original version of the file, copy and paste it and commit again?

**Comment by andrewcraik** [comment]
So one way to do this would be to do something like `git show some_commit_sha1 -- some_file.c | git apply -R` and then commit --amend... You just need to make sure you have no changes in your commit other than the changes you intended to make. You will also want to rebase this forward since it has been a little while and doing so will simplify running tests on your change once you amend and repush your commit.

**Comment by andrewcraik** [comment]
@mrSeoD do you plan to fix this PR so we can look to get it merged or shall we close it if you are no longer interested in making your contribution? It would be great to have this item completed if you have time. Thank you again for your interest in contributing to the project!

**Comment by 0xdaryl** [comment]
This PR appears to have gone dormant.  I will close this issue now, but if you wish to continue to make progress with it please feel free to continue working on it in your fork and creating a new PR when it is ready.

**Pull request closed without merging**